src/test/java/org/jboss/xavier/integrations/route/MainRouteBuilder_DirectAddUsernameHeaderTest.java

```diff
@@ -103,7 +105,68 @@ public void mainRouteBuilder_routeDirectAddUsernameHeader_BadContentGiven_Should
         });
 
         //Then
-        assertThat(result.getIn().getHeader("analysisUsername")).isNull();
+        assertThat(result.getIn().getHeader(RouteBuilderExceptionHandler.USERNAME)).isNull();
+        camelContext.stop();
+    }
+
+    @Test
+    public void mainRouteBuilder_xRhIdentityHeaderProcessor_givenInvalidHeader_shouldSetNullIdentityHeader() throws Exception {
+        // Given
```

Comment:
if I dont understand wronly , you are testing a method *MainRouteBuilder.xRhIdentityHeaderProcessor* . So I dont see any point on creating a Camel context with only 1 route that executes that method.

Either you create a unit test to that method, or you test the real routes (starting only those affecting the new code).

Reply:
+1. I've removed the route that was created inside the test and use `camelContext.startRoute("add-username-header");` instead